fe/src/main/java/org/apache/doris/analysis/CreateTableStmt.java

```diff
@@ -362,6 +392,68 @@ public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
             }
             columns.add(col);
         }
+
+        if (CollectionUtils.isNotEmpty(indexDefs)) {
+            Set<String> distinct = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+            Set<List<String>> distinctCol = new HashSet<>();
+
+            for (IndexDef indexDef : indexDefs) {
+                indexDef.analyze();
+                if (engineName.equalsIgnoreCase("olap")) {
```

Comment:
```suggestion
                if (!engineName.equalsIgnoreCase("olap")) {
                    throw new Exception;
                }
```
which can make code less indentation